lib/ansible/modules/cloud/cloudstack/cs_storage_pool.py

```diff
@@ -0,0 +1,399 @@
+    provider: DefaultPrimary
+    name: Ceph RBD
+    scope: ZONE
+    hypervisor: KVM
+
+# Ensure a Cluster scoped storage_pool is present
+- local_action:
+    module: cs_storage_pool
+    zone: zone01
+    cluster: cluster01
+    storage_url: rbd://admin:SECRET@ceph-the-mons.domain/poolname
+    provider: DefaultPrimary
+    name: Ceph RBD
+    scope: CLUSTER
+
+# Ensure a storage_pool is absent
+- local_action:
+    module: cs_storage_pool
+    name: Ceph RBD
+    state: absent
+'''
+
+RETURN = '''
+---
+id:
+    description: UUID of the pool.
+    returned: success
+    type: string
+    sample: a3fca65a-7db1-4891-b97c-48806a978a96
+capacity_iops:
+    description: IOPS CloudStack can provision from this storage pool
+    returned: when available
+    type: int
+    sample: 60000
+zone_id:
+    description: UUID of the zone.
+    returned: success
+    type: string
+    sample: a3fca65a-7db1-4891-b97c-48806a978a96
+zone_name:
+    description: The name of the cluster.
+    returned: success
+    type: string
+    sample: Zone01
+cluster_id:
+    description: UUID of the cluster.
+    returned: when scope is cluster
+    type: string
+    sample: a3fca65a-7db1-4891-b97c-48806a978a96
+cluster_name:
+    description: The name of the cluster.
+    returned: when scope is cluster
+    type: string
+    sample: Cluster01
+pod_id:
+    description: UUID of the pod.
+    returned: when scope is cluster
+    type: string
+    sample: a3fca65a-7db1-4891-b97c-48806a978a96
+pod_name:
+    description: The name of the pod.
+    returned: when scope is cluster
+    type: string
+    sample: Cluster01
+disk_size_allocated:
+    description: The pool's currently allocated disk space
+    returned: success
+    type: int
+    sample: 2443517624320
+disk_size_total:
+    description: The total size of the pool
+    returned: success
+    type: int
+    sample: 3915055693824
+disk_size_used:
+    description: The pool's currently used disk size
+    returned: success
+    type: int
+    sample: 1040862622180
+scope:
+    description: "The scope of the storage pool [ZONE / CLUSTER]"
+    returned: success
+    type: string
+    sample: CLUSTER
+state:
+    description: The state of the storage pool
+    returned: success
+    type: string
+    sample: Up
+tags:
+    description: the Tags for the storage pool
+    returned: success
+    type: string
+    sample: rbd
+'''
+
+# import cloudstack common
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    CloudStackException,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackStoragePool(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackStoragePool, self).__init__(module)
+        self.returns = {
+            'id': 'id',
+            'capacityiops': 'capacity_iops',
+            'zonename': 'zone_name',
+            'zoneid': 'zone_id',
```

Comment:
OK. I know we don't really want to know it in ansible, but it might be useful for external scripting. I'll get rid of it.